test/test-PreviewProvider.js

```diff
@@ -50,6 +50,19 @@ const gMockMetadataStore = {
       });
     }
     return items;
+  },
+  asyncExecuteQuery(query) {
+    // extract key from the query
+    let key = query.split("=")[1].trim().replace(/'/g, "");
```

Comment:
Part of the reason why these test changes are odd is that the API exposes the expectation of handling any SQL query, so to be complete, the test here should actually test for `INSERT` and `SELECT .. JOIN .. GROUP BY .. ORDER BY ..` etc. If instead the API is that it takes a single string key and returns true/false; that's quite a bit easier to test.